packages/vite/src/node/plugins/resolve.ts

```diff
@@ -357,7 +360,7 @@ function tryFsResolve(
 
     if (
       (res = tryResolveFile(
-        file + ext,
+        fileWithoutJsExt + ext,
```

Comment:
I don't think this is the right fix. We should be updating here:

https://github.com/vitejs/vite/blob/2136f2bb960d1a81ac3b3ca04d9ebd89dba44661/packages/vite/src/node/plugins/resolve.ts#L439

and return multiple possible ts src paths, e.g. `.ts` and `.tsx` for a `.js` import. And try to resolve for each of them.

Reply:
I've update the code of `getTsSrcPath` to return an array of regex values and then try each item in the array. Let me know you thoughts - definitely open to any suggestions.